### DISCUSSION TITLE
Can elasticsearch be used as a data source?

### DISCUSSION THREAD

**Discussion by evannwang1996** [vectordotdev/vector#20094] [Q&A]
I would like to ask everyone, can vector support pulling data from the elasticsearch cluster, cleaning the data in the middle, and finally outputting it to the elasticsearch cluster?

**Comment by jszwedko** [comment]
Hi @evannwang1996 ,

Unfortunately not, see https://github.com/vectordotdev/vector/issues/10170 for the feature request. As a workaround you could use another process to read the data from Elasticsearch and forward it to Vector.

**Reply by evannwang1996** [reply]
Thanks for the reply. I have read this discussion carefully. He and I have the same meaning. We both want vector to support retrieving data from one elasticsearch cluster and then output it to another elasticsearch cluster. There seems to be no ready-made solution here. Maybe I can implement an elasticsearch source plugin like this? Is it feasible to do secondary development on vector’s current open source code?

**Reply by jszwedko** [reply]
There is no way to create a plugin for Vector at the moment, but you can write another program that reads from Elasticsearch and then forwards to Vector using one of the existing sources.
